Add backend

vihanb opened the pull request:
### I have:
 - [x] Ran a test (`npm run dev && npm test`)
 - [ ] Added tests for my changes
 - [x] Documented my code thoroughly.

vihanb committed:
Bassic backend

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Basic backend modification

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Added backend assignment compilation

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Fixed docs

Signed-off-by: Vihan B <contact@vihan.org>

vihanb commented:
Do we want to stream to `lli` for JIT as part of the backend? We can resolve using `` `llvm-config --bindir`/lli``

vihanb committed:
Adds verification pass for lone literals.

This ensures that no literals are entered by themselves on a line on a
top-level expression.

Closes #52

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Merge branch 'develop' into add-backend

vihanb committed:
Added webpack compatibility

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Added string + integer + decimal (repr) compile

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Add some new classes for backend features like []

Signed-off-by: Vihan B <contact@vihan.org>

somebody1234 committed:
Merge remote-tracking branch 'origin/develop' into add-backend

vihanb committed:
Add libvsl JS

Signed-off-by: vihanb <contact@vihan.org>

vihanb committed:
Move to new scope-system

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Merge remote-tracking branch 'refs/remotes/origin/add-backend' into add-backend

vihanb committed:
Fix error messages with filename

vihanb committed:
Fix error manager bug I think

vihanb committed:
Merge branch 'add-backend' of github.com:vsl-lang/VSL into add-backend

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Add compiler

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Add type lookup + major changes

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Finalize classes, typealias, type exprs & assignment

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Add function lookup

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Fix scoping

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Fix returnref/returntype mixup

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Update VSLNative

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Working templates

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Modify regen type

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Upgrade to babel 7

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Fix bug

vihanb committed:
Add backend names

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Merge branch 'add-backend' of github.com:vsl-lang/VSL into add-backend

vihanb committed:
Add JS backend

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Update some babel things

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Merge branch 'add-backend' of github.com:vsl-lang/VSL into add-backend

taylorhansen commented:
What's the purpose of this feature branch? It's hard to tell from those vaguely-worded commits.

vihanb committed:
Change a lot of things

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add function compilation, byte sequence,& fixes.

Fixes:
 - Query set issue (fixed #62)
 - ID resolver always true (fixed #63)

Adds:
 - Increased type support
 - Byte sequence literal
 - Pointer type support

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add test Crypto library using basics bindings.

Tests for the `bindgen` build step.

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Compiling backend supports basic compilation

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add pkg opts

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add annotation and other modifires.

This adds:
 - @inline
 - @private local behavior
 - Subscopes
 - Entry verification
 - String layout

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add TD/LR call resolver

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add function overloading

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Fix gitmodules

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add returns and alias backend support.

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add strings, more primitive types and bitcast.

This will support a wider variety of integers along with bitcasting
strings to the supposed type (at compile-time).

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add if statement compilation

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Bug fixes and tweaks

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Centralize code block generation.

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Fix more bugs in call resolver

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Fix babel dependencies

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add strings and property resolvers

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Made more browser-friendly

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add properties

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Fix more type resolver bugs.

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Add WASM and `vsl build` commands.

Adds compiler toolchain to quickly obtain VSL binaries. Additionally
adds a VSL WASM support which allows you to compile VSL WASM binaries
which you can interface with by using `vsl/wasm`

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Trim bulk and better WASM support

Signed-off-by: Vihan <contact@vihan.org>

vihanb committed:
Fix default to properly execute code.

Signed-off-by: Vihan <contact@vihan.org>
